src/controllers/search-product-name.controller.js

```diff
@@ -0,0 +1,22 @@
+const searchProductService = require("../services/search-product-name.service");
```

Comment:
```suggestion
const productService = require("../services/product.service");
```
해당 api 기능 자체가 상품을 기반으로 검색하는 것이고, 이는 상품과 관련된 layer라는 것을 의미하므로 product layer안에 search와 관련된 기능을 function으로 들어가면 됩니다. 지금 같은 상황이라면 똑같이 상품에 대한 api인데, `search-product-name.service`, `filter-product-name.service` 등등 불필요하게 분리되어 복잡한 패턴을 갖게 될 것  같아요.